client/src/controllers/BlockController.ts

```diff
@@ -0,0 +1,83 @@
+import { Controller } from '@hotwired/stimulus';
+import { object } from 'prop-types';
+
+declare global {
+  interface Window {
+    initBlockWidget?: (id: string) => void; // Declare the initBlockWidget function as optional on the Window interface
+    telepath: any;
+  }
+}
+
+/**
+ * Adds the ability to unpack a Telepath object and render it on the controlled element.
+ *
+ * @example
+ * <div
+ * id="{id}"
+ * data-block
+ * data-controller="w-block"
+ * data-w-block-data-value="{block_json}"
+ * data-w-block-initial-value="{value_json}"
+ * data-w-block-error-value="{error_json}">
+ * </div>
+ */
+
+export class BlockController extends Controller<HTMLElement> {
+  /**
+   * initial - Initial value for the block
+   * data - Unpacked Telepath data
+   * error - Block error (optional)
+   */
+  static values = {
+    initial: { type: object, default: undefined },
+    data: { type: object, default: undefined },
+    error: { type: object, default: undefined },
+  };
+
+  /** Packed Telepath data. */
+  declare dataValue: object;
+  declare errorValue: object;
+  declare initialValue: object;
```

Comment:
Just a clean up, we still need lowercase `object` here but I suggest we re-order this to alpha and add a basic JSDoc to each.

```suggestion
  /** Initial value. */
  declare initialValue: object;
  /** Optional error value. */
  declare errorValue: object;
  /** Block data value to unpack with Telepath. */
  declare dataValue: object;
```